File: Part 1 - Data Preprocessing/DataPreprocessing_Template.R
#Data Processing

#Importing dataset
dataset = read.csv("Data.csv")
#dataset = dataset[,2:3]

#Take care of missing data
dataset$Age = ifelse(is.na(dataset$Age),
                ave(dataset$Age, FUN = function(x) mean(x,na.rm = TRUE)),
                dataset$Age)

dataset$Salary = ifelse(is.na(dataset$Salary),
                     ave(dataset$Salary, FUN = function(x) mean(x,na.rm = TRUE)),
                     dataset$Salary)

#Encoding categorical data
dataset$Country = factor(dataset$Country,
                         levels=c('France','Spain','Germany'),
                         labels=c(1,2,3))

dataset$Purchased = factor(dataset$Purchased,
                         levels=c('No','Yes'),
                         labels=c(0,1))

#Splitting data into training and test set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_Set = subset(dataset, split == TRUE)
test_Set = subset(dataset, split == FALSE)

#Feature Scaling
training_Set[,2:3] = scale(training_Set[,2:3])
test_Set[,2:3] = scale(test_Set[,2:3])

File: Part 1 - Data Preprocessing/DataPreprocessing_Template.py
#Data Preprocessing

#Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing dataset
dataset = pd.read_csv("Data.csv")
X = dataset.iloc[:,:-1].values
Y = dataset.iloc[:,3].values


#Taking care of missing data
from sklearn.preprocessing import Imputer 
imputer = Imputer(missing_values='NaN', strategy='mean', axis=0)
imputer = imputer.fit(X[:,1:3])
X[:,1:3] = imputer.transform(X[:,1:3])

#Encode categorical data
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X = LabelEncoder()
X[:,0] = labelencoder_X.fit_transform(X[:,0])
onehotencoder = OneHotEncoder(categorical_features=[0])
X = onehotencoder.fit_transform(X).toarray(111111)

labelencoder_Y = LabelEncoder()
Y = labelencoder_Y.fit_transform(Y)

#Split the data into training set and test set
from sklearn.cross_validation import train_test_split
X_Train, X_Test, Y_Train, Y_Test = train_test_split(X, Y, test_size=0.2, random_state=0 )

#Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_Train = sc_X.fit_transform(X_Train)
X_Test = sc_X.transform(X_Test)